eaf-pdf-viewer.el

```diff
@@ -56,69 +56,57 @@
 ;; No need more.
 
 ;;; Customize:
-;;
-;;
-;;
-;; All of the above can customize by:
-;;      M-x customize-group RET eaf-pdf-viewer RET
-;;
-
-;;; Change log:
-;;
-;; 2021/07/20
-;;      * First released.
-;;
-
-;;; Acknowledgements:
-;;
-;;
-;;
-
-;;; TODO
-;;
-;;
-;;
-
-;;; Require
-
-
-;;; Code:
 
-(defvar eaf-pdf-outline-buffer-name "*eaf pdf outline*"
-  "The name of pdf-outline-buffer.")
-
-(defvar eaf-pdf-outline-window-configuration nil
-  "Save window configure before popup outline buffer.")
+(defgroup eaf-pdf-viewer nil
+  "The PDF viewer application of Emacs application framework."
+  :group 'eaf)
 
 (defcustom eaf-pdf-extension-list
   '("pdf" "xps" "oxps" "cbz" "epub" "fb2" "fbz")
   "The extension list of pdf application."
-  :type 'cons)
+  :type 'list
+  :group 'eaf-pdf-viewer)
 
 (defcustom eaf-office-extension-list
   '("docx" "doc" "ppt" "pptx" "xlsx" "xls")
   "The extension list of office application."
-  :type 'cons)
+  :type 'list
+  :group 'eaf-pdf-viewer)
 
 (defcustom eaf-pdf-store-history t
   "If it is t, the pdf file path will be stored in eaf-config-location/pdf/history/log.txt for eaf-open-pdf-from-history to use"
-  :type 'boolean)
+  :type 'boolean
+  :group 'eaf-pdf-viewer)
 
 (defcustom eaf-pdf-dark-mode "follow"
-  ""
-  :type 'string)
+  "Whether to enable inverted color rendering when starting the pdf-viewer app.
+
+Possible values are
+   - \"follow\" Follow the background color of the theme of user.
+   - \"force\" Force inverted color rendering on start-up.
+   - \"ignore\" Don't do inverted rendering."
+  :type '(choice (string :tag "Force inverted color rendering." "force")                 
```

Comment:
Please remove whitespace of line end.

I think , `ignore` is duplicate, if user not setting `eaf-pdf-dark-mode` or setting it to `""`, will default is normal.

Reply:
I don't quite understand what you've said. I thought `"ignore"` meant disabling the dark mode altogether, do you want me to remove that as the "other" choice in customize interface?

Reply:
> Possible values are
>   - \"follow\" Follow the background color of the theme of user.
>   - \"force\" Force inverted color rendering on start-up.
>   - \"ignore\" Don't do inverted rendering."

OK, I understand your meant. 

At first, I looked at this document and thought I could only choose one of the above three.(I misunderstand it)